trainer.py

```diff
@@ -62,16 +62,15 @@ def train_one_epoch(self) -> tuple[float, float]:
         for batch_idx, (inputs, targets) in enumerate(train_dataloader):
             inputs, targets = inputs.to(device), targets.to(device)
 
-            self.optimizer.zero_grad()
             predictions = self.model(inputs)
             loss = self.criterion(predictions, targets)
+            self.optimizer.zero_grad()
             loss.backward()
             self.optimizer.step()
 
             correct_labeled_samples += (predictions.argmax(1) == targets).type(torch.int).sum().item()
-            nof_samples += len(inputs)
             total_loss += loss.item()
-            avg_loss = total_loss / nof_samples
+            avg_loss = total_loss / (batch_idx + 1)
             accuracy = correct_labeled_samples * 100 / nof_samples
```

Comment:
should be:
accuracy = correct_labeled_samples * 100 / ((batch_idx + 1) * self.batch_size))

you didnt go over all the samples yet so its unfair to divide by the entire set